src/gui/widget/TextField.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef PUTKARTS_GUI_Widget_TextField_HPP
+#define PUTKARTS_GUI_Widget_TextField_HPP
+
+#include <string>
+
+#include "gui/widget/Widget.hpp"
+
+#include "SFML/Graphics.hpp"
+
+namespace GUI {
+	namespace Widget {
+		class TextField;
+	}
+}
+
+/**
+ * One row textfield.
+ */
+class GUI::Widget::TextField: public Widget {
+	/** The text */
+	sf::Text text;
```

Comment:
How about using std::string for storage and a Label for display, maybe extending the Label a bit if needed?